src/mechanics/elements/SpectralShapeFunctions.h

```diff
@@ -0,0 +1,248 @@
+//! @param x local coordinate where shapes are evaluated
+//! @param nodes local node coordinates
+//! @return shapes evaluated at x. Fulfill interpolation condition fi(xj) = delta_ij
+std::vector<double> ShapeFunctionsQuadLagrange(const Eigen::Vector2d x, const std::vector<double>& nodes)
+{
+    std::vector<double> Nx = ShapeFunctions1D::ShapeFunctionsTrussLagrange(x(0), nodes);
+    std::vector<double> Ny = ShapeFunctions1D::ShapeFunctionsTrussLagrange(x(1), nodes);
+
+    std::vector<double> result;
+    for (size_t j=0; j<nodes.size(); j++)
+    {
+        for (size_t i=0; i<nodes.size(); i++)
+        {
+            result.push_back(Nx[i] * Ny[j]);
+        }
+    }
+    return result;
+}
+
+std::vector<Eigen::Vector2d> DerivativeShapeFunctionsQuadLagrange(const Eigen::Vector2d x, const std::vector<double>& nodes)
+{
+    std::vector<double> DNx = ShapeFunctions1D::DerivativeShapeFunctionsTrussLagrange(x(0), nodes);
+    std::vector<double> DNy = ShapeFunctions1D::DerivativeShapeFunctionsTrussLagrange(x(1), nodes);
+
+    std::vector<double> Nx = ShapeFunctions1D::ShapeFunctionsTrussLagrange(x(0), nodes);
+    std::vector<double> Ny = ShapeFunctions1D::ShapeFunctionsTrussLagrange(x(1), nodes);
+
+    std::vector<Eigen::Vector2d> result;
+    for (size_t j=0; j<nodes.size(); j++)
+    {
+        for (size_t i=0; i<nodes.size(); i++)
+        {
+            result.push_back( Eigen::Vector2d({DNx[i]*Ny[j] , Nx[i]*DNy[j]})   );
+        }
+    }
+    return result;
+}
+
+}
+
+
+
+namespace ShapeFunctions3D
+{
+//! @brief Lobatto nodes for brick element (tensor product)
+//! @param order Polynomial order in 1D
+//! @return local node coordinates
+std::vector<Eigen::Vector3d> NodeCoordinatesBrickLobatto(int order)
+{
+    std::vector<double> nodes = ShapeFunctions1D::NodeCoordinatesTrussLobatto(order);
+    std::vector<Eigen::Vector3d> result;
+    for (size_t i=0; i<nodes.size(); i++)
+    {
+        for (size_t j=0; j<nodes.size(); j++)
+        {
+            for (size_t k=0; k<nodes.size(); k++)
+            {
+                result.push_back(Eigen::Vector3d({nodes[i],nodes[j],nodes[k]}));
+            }
+        }
+    }
+    return result;
+}
+
+//! @brief Value of Lagrange type shape functions with given 1D nodes (tensor product)
+//! @param x local coordinate where shapes are evaluated
+//! @param nodes local node coordinates
+//! @return shapes evaluated at x. Fulfill interpolation condition fi(xj) = delta_ij
+std::vector<double> ShapeFunctionsBrickLagrange(const Eigen::Vector3d x, const std::vector<double>& nodes)
+{
+    std::vector<double> Nx = ShapeFunctions1D::ShapeFunctionsTrussLagrange(x(0), nodes);
+    std::vector<double> Ny = ShapeFunctions1D::ShapeFunctionsTrussLagrange(x(1), nodes);
+    std::vector<double> Nz = ShapeFunctions1D::ShapeFunctionsTrussLagrange(x(2), nodes);
+
+    std::vector<double> result;
+    for (size_t k=0; k<nodes.size(); k++)
```

Comment:
Here, the loop order is `k -- j --i`, above it is `i -- j -- k`. Is that important?

Reply:
Here (Shapes) the order is important (k-j-i), above (NodeCoordinates) it doesnt matter - but I think I will change it also to k-j-i for better reading.